codegen/core/src/main/scala/com/lightbend/akkasls/codegen/Packaging.scala

```diff
@@ -114,11 +127,20 @@ object PackageNaming {
     val javaPackage =
       generalOptions.get("google.protobuf.FileOptions.java_package").map(_.toString())
 
-    val javaOuterClassname =
+    val javaOuterClassnameOption =
       generalOptions
         .get("google.protobuf.FileOptions.java_outer_classname")
         .map(_.toString())
 
+    val javaOuterClassname =
+      javaOuterClassnameOption.getOrElse {
+        val existingNames =
+          descriptor.getMessageTypes.asScala.map(_.getName) ++
+          descriptor.getEnumTypes.asScala.map(_.getName) ++
+          descriptor.getServices.asScala.map(_.getName)
+        if (existingNames.contains(name)) name + "OuterClass" else name
+      }
```

Comment:
Second part of the fix, check whether `OuterClass` will be added to the name.

Reply:
My memory is failing me. Is that the default behavior when generating java classes from protobuf? 

This is how I read the current PR: 

* if multiple-files = true, each type has it's one file. No wrapper, all is fine.
* if needs a wrapper, the wrapper defaults to file name.
* if name generated from the file name conflicts with service name, wrapper gets `OuterClass` appended to it.

Anyway, I'm fine with it. I'm just wondering if that's an existing convention.

Reply:
Yes, appending `OuterClass` when it is needed for disambiguation is a protoc-java convention, i.e. https://github.com/akka/akka-grpc/commit/8b39a4098f6a12bce2b909f4b94fb00424f63d85 and https://github.com/protocolbuffers/protobuf/blob/4e0a1119c0c5cdfe89b54b9d66cb76223f17861a/src/google/protobuf/compiler/java/java_name_resolver.cc#L157

Reply:
thanks for clarifying

Reply:
Yes, should be aligned with the protobuf generation now. And we also still use the outer class for accessing the descriptor (whether or not it's multiple files enabled), so we need to have the right name for that.